### ISSUE TITLE
Unmatched quote in a #warning pragma causes MA0115 false positive

### ISSUE THREAD

**Issue by wizardness** [meziantou/Meziantou.Analyzer#1243]
### Version of the Meziantou.Analyzer NuGet package

3.0.132

### Rule Identifier

MA0115

### Target Framework

.NET 10

### C# Language version

C# 14.0

### Description

If there is an unmatched " or ' inside a #warning pragma in a child component, the analyzer can't see any of the child's parameters.

### Reproduction Steps

When a #warning in this component contains an unmatched ' or ":
```c#
@code {
    // WarningGuy.razor
    [Parameter] public string Foo { get; set; } = "";

    protected override void OnAfterRender(bool firstRender)
    {
#warning This warning is fine.
#warning "  This warning causes MA0115 to miss the Foo declaration.
// #warning '  This warning causes MA0115 to miss the Foo declaration.

        base.OnAfterRender(firstRender);
    }
}

```

This component triggers MA0115 
```c#
<h3>ConfusedMA0115 </h3>

<WarningGuy Foo="bar"/>
```

The Confused component won't see any parameters in the child component.

**Comment by meziantou** [comment]
I do not reproduce the error locally with .NET 11 preview 6.
- Which version of .NET SDK are you using?
- Which Target Framework?
- Can you share a small repro project?

**Comment by Frulfump** [comment]
> I do not reproduce the error locally with .NET 11 preview 6.
> 
>     * Which version of .NET SDK are you using?
> 
>     * Which Target Framework?
> 
>     * Can you share a small repro project?

Seems like TFM was provided in the description. I noticed that the bug issue template lacks version of .NET SDK should that be added there as well?

**Comment by meziantou** [comment]
> Seems like TFM was provided in the description. I noticed that the bug issue template lacks version of .NET SDK should that be added there as well?

Thanks for the suggestion! I've just added it. Note that it's not needed most of the time, but here the Blazor generator may be relevant.

**Comment by wizardness** [comment]
- Which version of .NET SDK are you using?
-- 10.0.300

* Which Target Framework?
-- net10.0

* Can you share a small repro project?
[Issue1234.zip](https://github.com/user-attachments/files/30464976/Issue1234.zip)

Thank you for taking a look!

**Comment by meziantou** [comment]
I still don't repro. I've added a global.json file to force `10.0.302` (latest version). Can you update the SDK to the latest version and can you share the output of `dotnet build`? On my side:

```
Restore complete (0.2s)
  Issue1243 net10.0 succeeded with 2 warning(s) (0.2s) → bin/Debug/net10.0/Issue1243.dll
    /Users/meziantou/Downloads/Issue1234/WarningGuy.razor(9,10): warning CS1030: #warning: 'This warning is fine.'
    /Users/meziantou/Downloads/Issue1234/WarningGuy.razor(10,10): warning CS1030: #warning: '"  This warning causes MA0115 to miss the Foo declaration.'

Build succeeded with 2 warning(s) in 0.6s
```

The 2 warnings are expected because of the `#warning`

**Comment by wizardness** [comment]
Dangit, I forgot the actual important bit!

The problem manifests in JetBrains Rider's "Problems" pane, which probably means an entirely different intermediate code generator.  The build itself performs as expected.

Oddly, the Rider context highlighting and mouse-over correctly reports that Foo is a valid parameter and (newly-added) Baz is not:
<img width="895" height="178" alt="Image" src="https://github.com/user-attachments/assets/da85bb7b-30c5-4407-afe2-c4e971455bcb" />

I tried to repeat it in VS Code today and failed.  

(It is more than fair to call this a "won't fix" priority as it's very limited and I can just remove the quotes.)

**Comment by wizardness** [comment]
In a fit of ignorant flailing, I did try dumping Rider's PSI code-behind files and saw nothing obvious.  (Tools -> Internal Actions (Resharper) -> PSI -> Dump Code-behind)

Just in case:

[ConfusedMA0115.Zyqitot.cs](https://github.com/user-attachments/files/30515388/ConfusedMA0115.Zyqitot.cs)

[WarningGuy.Padizuz.cs](https://github.com/user-attachments/files/30515401/WarningGuy.Padizuz.cs)

**Comment by meziantou** [comment]
If you don't get any warning when compiling the project using the CLI (`dotnet build`), but only get the warning in Rider, then please close this issue and open one in JetBrains.

**Comment by wizardness** [comment]
https://youtrack.jetbrains.com/issue/RIDER-141469/Unmatched-quote-in-warning-causes-false-positive-analysis-problem-Blazor
